code/script.js

```diff
@@ -1,47 +1,288 @@
+
+// Space-Bot starts here
 
-// Global variables, if you need any, declared here
+/*Typing effect  */
+let i = 0;
+let txt = "";
+let textAnimationContainer;
+const speed = 50; 
 
-// Functions declared here
+const typeReply = () => {
+  if (i < txt.length) {
+    textAnimationContainer.innerHTML += txt.charAt(i);
+    i++;
+    setTimeout(typeReply, speed);
+  }
+}
 
 // This function will add a chat bubble in the correct place based on who the sender is
 const showMessage = (message, sender) => {
   if (sender === 'user') {
-    chat.innerHTML += `
-      <section class="user-msg">
-        <div class="bubble user-bubble">
-          <p>${message}</p>
-        </div>
-        <img src="assets/user.png" alt="User" />  
-      </section>
-    `
+  const section = document.createElement('section');
+  section.classList.add("user-msg");
+  chat.appendChild(section);
+  section.innerHTML += 
+  `<div class="bubble user-bubble">
+  <p></p>
+  </div>
+  <img src="assets/avatar.png" alt="User" /> `;
+  textAnimationContainer = section.querySelectorAll('p')[0];
+  txt = message;
+  i=0;
+  typeReply();
+
   } else if (sender === 'bot') {
     chat.innerHTML += `
-      <section class="bot-msg">
-        <img src="assets/bot.png" alt="Bot" />
-        <div class="bubble bot-bubble">
-          <p>${message}</p>
-        </div>
-      </section>
-    `
+    <section class="bot-msg">
+      <img src="assets/alien-img.png" alt="Bot" />
+      <div class="bubble bot-bubble">
+        <p>${message}</p>
+      </div>
+    </section>
+  `;
   }
-  // This little thing makes the chat scroll to the last message when there are too many to be shown in the chat box
-  chat.scrollTop = chat.scrollHeight
+// This little thing makes the chat scroll to the last message when there are too many to be shown in the chat box
+setTimeout (() => chat.scrollTop = chat.scrollHeight, 100);
 }
 
 // Starts here
+// Question 1
 const greeting = () => {
-  showMessage(`Hello there, What's your name?`, 'bot')
-  // Just to check it out, change 'bot' to 'user' here 👆
+  showMessage(`You wanna go to Space? Cool! What's your name?`, 'bot');
+}
+
+handleNameInput.addEventListener('submit', (event) => {
+  event.preventDefault();
+  const nameInput = document.getElementById('name-input');
+  const userName = nameInput.value;
+  userFormInputs.name = userName;
+  
+  if (!isNaN(nameInput.value)) {
+    nameInput.value = '' ;
+    alert("You can't have numbers in you Space-name"); 
+  }
+  else {
+    showMessage (userName, 'user');
+    nameInput.value = '' ;
+    setTimeout(() => spaceAgeQuestion(userName), 1000);
+    
+  }
+}
+)
+
+// Question 2
+const spaceAgeQuestion = (userName) => {
+  showMessage (`Hi ${userName}! Time goes slower in space. How many Earth years do you want to spend in Space?`, 'bot');
+  
+    inputWrapper.innerHTML = `
+    <button  class="bot-button" id="oneYear">1 year</button>
+    <button  class="bot-button" id="tenYears">10 years</button>
```

Comment:
you used different input styles in your projects which make it more dynamic, gg!